src/arch/xtensa/boot_loader.c

```diff
@@ -93,6 +115,36 @@ static void parse_module(struct sof_man_fw_header *hdr,
 #define MAN_SKIP_ENTRIES 1
 #endif
 
+static uint32_t  get_fw_size_in_use(void)
+{
+	struct sof_man_fw_desc *desc =
+		(struct sof_man_fw_desc *)MANIFEST_BASE;
+	struct sof_man_fw_header *hdr = &desc->header;
+	struct sof_man_module *mod;
+	uint32_t fw_size_in_use = 0xffffffff;
+	int i;
+
+	/* Calculate fw size passed in BASEFW module in MANIFEST */
+	for (i = MAN_SKIP_ENTRIES; i < hdr->num_module_entries; i++) {
+		platform_trace_point(TRACE_BOOT_LDR_PARSE_MODULE + i);
+		mod = (void *)desc + SOF_MAN_MODULE_OFFSET(i);
+		if (~strcmp((char *)mod->name, "BASEFW")) {
+
```

Comment:
You might want to use ! instead of ~.

Reply:
I do not like that construct - i think it makes code harder to read

Reply:
@dabekjakub The idea of ending a loop or function early is actually rather prevalent in both Linux and SOF. Instead of positive checks, you do a negative check and if the negative check succeeds you either end the loop iteration (in our case the `continue;` idea suggested by @dbaluta ), the loop itself, do something like `goto error;` (at least in Linux I don't think there is any driver where this is _not_ used) or end the function with a `return`.

All of these help with indentation (given the rather stringent limits of about 80 characters per line, such constructs are often necessary to fit in them). The only reason I've encountered that allows me to do more than that was to not break a string.

Also, try to adhere to https://www.kernel.org/doc/html/v4.10/process/coding-style.html (SOF uses the same coding style rules as Linux, for the most part)

Reply:
Where exactly did my code stray from guidelines @paulstelian97 
Many things are prevalent. Which doesn't mean they are good.
I changed it to what You want - however this artificial constraints are making work hard...

Reply:
@dabekjakub You need to align to what others are doing really. And this one you'll get used to. Anyway I've encountered one situation where this rule isn't obeyed and it was actually harder to understand. At least for me, if not for others too, it's easier to understand getting the cases where you shouldn't do anything out of the way by this fashion.

At least your situation didn't also have duplication.